xbmc/Application.cpp

```diff
@@ -4865,6 +4865,13 @@ bool CApplication::OnMessage(CGUIMessage& message)
       if (url.IsProtocol("plugin"))
         XFILE::CPluginDirectory::GetPluginResult(url.Get(), file);
 
+      // Don't queue if next media type is different from current one
+      if ((!file.IsVideo() && m_pPlayer->IsPlayingVideo()) ||
```

Comment:
>could you elaborate on the intent here?

Support mixed media playlists(i.e playlists with both music and video). In xbmc 12 and before, it worked without problems, but since the inclusion of GUI_MSG_QUEUE_NEXT_ITEM, PAPlayer would skip videos it encounters in the playlist. The idea here is ignore the QUEUE message and just act as the playback already finished to allow Kodi to find a new suitable player for the media in the list. 

> assume audio and video files are directed to dvdplayer which can handle both types. why exit here?

DVD player doesn't have queuing support, PAPlayer works much nicer if you have more than two songs in a row, so it seems a better idea to return to it if when a song appears.

Anyway, an alternative patch for that line could be:
```if (file.IsVideo() && !m_pPlayer->HasVideo()) ....```
That would only exit if using PAPlayer and a video is in the list, but won't return to PAPlayer if playing video and a song comes.